src/main/java/com/mitchej123/hodgepodge/mixins/minecraft/MixinNetHandlePlayClient.java

```diff
@@ -0,0 +1,36 @@
+package com.mitchej123.hodgepodge.mixins.minecraft;
+
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.network.NetHandlerPlayClient;
+import net.minecraft.network.play.server.S02PacketChat;
+import net.minecraft.util.ChatComponentTranslation;
+import net.minecraftforge.client.event.ClientChatReceivedEvent;
+import net.minecraftforge.common.MinecraftForge;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Overwrite;
+import org.spongepowered.asm.mixin.Shadow;
+
+@Mixin(value = NetHandlerPlayClient.class)
+public class MixinNetHandlePlayClient {
+
+    @Shadow
+    private Minecraft gameController;
+
+    /**
+     * @author Quarri6343
+     * @reason Stop "You can only sleep at night" message filling the chat
+     */
+    @Overwrite
+    public void handleChat(S02PacketChat p_147251_1_) {
+        ClientChatReceivedEvent event = new ClientChatReceivedEvent(p_147251_1_.func_148915_c());
+        if (!MinecraftForge.EVENT_BUS.post(event) && event.message != null) {
+            if (event.message.equals(new ChatComponentTranslation("tile.bed.noSleep", new Object[0]))
+                    || event.message.equals(new ChatComponentTranslation("tile.bed.notSafe", new Object[0]))
+                    || event.message.equals(new ChatComponentTranslation("tile.bed.occupied", new Object[0]))) {
+                this.gameController.ingameGUI.getChatGUI().printChatMessageWithOptionalDeletion(event.message, 3);
```

Comment:
imo you shouldn't use the printChatMessageWithOptionalDeletion method since you don't know which other mod uses which message ids, and it might start deleting other chat messages that you don't expect. 

If you want to delete chat messages you should make your own deleteChatLine method that iterates though `    private final List chatLines = new ArrayList();
    private final List field_146253_i = new ArrayList();`

like I did here https://github.com/Alexdoru/MegaWallsEnhancements/blob/4055c0d6434914bfeb89019d0299a2a85cea3a92/src/main/java/fr/alexdoru/megawallsenhancementsmod/asm/hooks/GuiNewChatHook.java#L26

Reply:
can I just set the message id a very big number?
there are 2 billion channels so it should not cause conflicts if I set it like 251545647

Reply:
![image](https://user-images.githubusercontent.com/57050655/188920758-3084ae8d-2f07-4836-a1a8-00a3b585ebd9.png)

Reply:
> can I just set the message id a very big number? there are 2 billion channels so it should not cause conflicts if I set it like 251545647

Imperfect but sounds reasonable

Reply:
Just stumbed upon this https://github.com/GTNewHorizons/EnderCore/blob/68ba8bc29db0bd00a344cb59fb3c25242e24c480/src/main/java/com/enderio/core/common/util/ChatUtil.java#L61

I guess we are fine with this solution

Reply:
Now we have HUD overlay above hotbar (https://github.com/GTNewHorizons/GTNHLib/pull/7), so we can use that I guess